airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/write/transform/PostgresValueCoercer.kt

```diff
@@ -76,9 +76,18 @@ class PostgresValueCoercer : ValueCoercer {
                 }
             }
             is StringValue -> {
+                // PostgreSQL doesn't allow null bytes (\u0000) in text fields
+                // Replace them with empty string to prevent COPY errors
+                // Using replace() without regex for optimal performance (O(n) vs O(n*m) with regex)
+                if (abValue.value.contains('\u0000')) {
+                    val sanitizedValue = abValue.value.replace("\u0000", "")
+                    value.abValue = StringValue(sanitizedValue)
```

Comment:
how do we decide that this is the behavior we want? as opposed to doing nullify like we do elsewhere?